opentelemetry/src/main/scala/zio/telemetry/opentelemetry/baggage/Baggage.scala

```diff
@@ -138,12 +138,28 @@ object Baggage {
             )
 
         override def set(name: String, value: String)(implicit trace: Trace): UIO[Unit] =
+          unclosedScope(setScoped(name, value))
+
+        private def setScoped(name: String, value: String)(implicit trace: Trace): URIO[Scope, Unit] =
           injectLogAnnotations *> modifyBuilder(_.put(name, value)).unit
 
+        // to preserve the existing behavior where the otel Scope returned from Context.makeCurrent is not closed
+        // see io.opentelemetry.context.Context.makeCurrent
+        private def unclosedScope[A](io: URIO[Scope, A]): UIO[A] =
```

Comment:
@grouzen this seems to me the way to preserve existing behaviour. What do you think?

Also: perhaps good to link to an issue in that comment?

Reply:
It needs to be checked with some end-to-end test. One of the options could be a manual verification using an optelemetry-instrumentation example. Additionally, it would be good to check how it works with dd-trace-java because it is the reason why this issue was created in the first place.

Reply:
I'm just afraid we close the current span this way, and I believe this is not what we want to do while modifying baggage data, right? That's why I think we need to do manual tests. Also, adding a unit test to prove/disprove my assumption would be great.

Reply:
It can't be fixed this way, as I said before. I think the whole idea of having specific ZIO2 instrumentation in opentelemetry-java-instrumentation and dd-trace-java was a mistake, hence the current implementation of ContextStorage.native was a mistake too. It must be done differently and I'm working on this right now.

Reply:
I think the better solution in your situation would be to use your fork with the fix for your case while I'm figuring out how to fix it properly to make everyone happy.

Reply:
Using the previous supervisor based approach?